File: leetcode/cpp/242. 有效的字母异位词.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }
        std::map<char, int> map;
        for (auto & c : s) {
            map[c] += 1;
        }
        for (auto & c : t) {
            map[c] -= 1;
            if (map[c] == -1) {
                return false;
            }
        }
        return true;
    }
};

File: leetcode/python/242. 有效的字母异位词.py
#!/bin/bash
"""
242. 有效的字母异位词

思路：
hash，或者使用char数组模拟hash
"""

class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        哈希
        """
        if len(s) != len(t): # 长度不等直接退出
            return False
        d = dict()
        for char in s:
            d[char] = 1 + d.get(char, 0)
        for char in t:
            if char in d:
                d[char] -= 1
            if char not in d or d[char] == -1:
                return False
        return True